ui/imports/shared/controls/TransactionDelegate.qml

```diff
@@ -9,187 +9,502 @@ import StatusQ.Controls 0.1
-        loading: root.loading
+        bgColor: "transparent"
+        color: Theme.palette.black
+        bgBorderWidth: 1
+        bgBorderColor: Theme.palette.primaryColor3
     }
-    TextMetrics {
-        id: textMetrics
-        font: statusListItemSubTitle.font
-        text: statusListItemSubTitle.text
+
+    sensor.children: [
+        StatusRoundIcon {
+            id: leftIconStatusIcon
+            visible: !root.loading
+            anchors {
+                right: root.statusListItemIcon.right
+                bottom: root.statusListItemIcon.bottom
+            }
+            asset: root.statusIconAsset
+        }
+    ]
+
+    // Title
+    title: {
+        if (root.loading) {
+            return "dummmy"
+        } else if (!root.isModelDataValid) {
+            return ""
+        }
+
+        const isPending = root.transactionStatus === TransactionDelegate.TransactionStatus.Pending
+        const failed = root.transactionStatus === TransactionDelegate.TransactionStatus.Failed
+        switch(root.transactionType) {
+        case TransactionDelegate.TransactionType.Send:            
+            return failed ? qsTr("Send failed") : (isPending ? qsTr("Sending") : qsTr("Sent"))
+        case TransactionDelegate.TransactionType.Receive:
+            return failed ? qsTr("Receive failed") : (isPending ? qsTr("Receiving") : qsTr("Received"))
+        case TransactionDelegate.TransactionType.Buy:
+            return failed ? qsTr("Buy failed") : (isPending ? qsTr("Buying") : qsTr("Bought"))
+        case TransactionDelegate.TransactionType.Sell:
+            return failed ? qsTr("Sell failed") : (isPending ? qsTr("Selling") : qsTr("Sold"))
+        case TransactionDelegate.TransactionType.Destroy:
+            return failed ? qsTr("Destroy failed") : (isPending ? qsTr("Destroying") : qsTr("Destroyed"))
+        case TransactionDelegate.TransactionType.Swap:
+            return failed ? qsTr("Swap failed") : (isPending ? qsTr("Swapping") : qsTr("Swapped"))
+        case TransactionDelegate.TransactionType.Bridge:
+            return failed ? qsTr("Bridge failed") : (isPending ? qsTr("Bridging") : qsTr("Bridged"))
+        default:
+            return ""
+        }
     }
-    components: [
-        ColumnLayout {
-            visible: !root.isNFT
-            Row {
-                Layout.alignment: Qt.AlignRight
-                spacing: 4
-                StatusIcon {
-                    color: isIncoming ? Theme.palette.successColor1 : Theme.palette.dangerColor1
-                    icon: "arrow-up"
-                    rotation: isIncoming ? 135 : 45
-                    height: 18
-                    visible: !root.loading
+    statusListItemTitleArea.anchors.rightMargin: root.rightPadding
+    statusListItemTitle.font.weight: Font.DemiBold
+    statusListItemTitle.font.pixelSize: root.loading ? d.loadingPixelSize : d.titlePixelSize
+
+    // title icons and date
+    statusListItemTitleIcons.sourceComponent: Row {
```

Comment:
Why the double `Row`?

Reply:
In first row we have Token images row and then timestamp text.
Second row is used with minus spacing because swap token images overlap horizontally.

Reply:
Second Row uses negative spacing because for Swap transaction token images are overlapping.
Using one Row would break timestamp text spacing.